pkg/odo/cli/pipelines/bootstrap.go

```diff
@@ -67,17 +70,32 @@ func (bo *BootstrapOptions) Validate() error {
 // Run runs the project bootstrap command.
 func (bo *BootstrapOptions) Run() error {
 	options := pipelines.BootstrapOptions{
-		DeploymentPath:   bo.deploymentPath,
-		GithubToken:      bo.githubToken,
-		GitRepo:          bo.gitRepo,
-		Prefix:           bo.prefix,
-		QuayAuthFileName: bo.quayIOAuthFilename,
-		QuayUserName:     bo.quayUsername,
-		SkipChecks:       bo.skipChecks,
+		DeploymentPath:           bo.deploymentPath,
+		GithubToken:              bo.githubToken,
+		GitRepo:                  bo.gitRepo,
+		ImageRepo:                bo.imageRepo,
+		InternalRegistryHostname: bo.internalRegistryHostname,
+		Prefix:                   bo.prefix,
+		QuayAuthFileName:         bo.quayIOAuthFilename,
+		QuayUserName:             bo.quayUsername,
+		SkipChecks:               bo.skipChecks,
+	}
+	if err := checkOptionalRegistryParams(options); err != nil {
+		return err
```

Comment:
if they're not optional, maybe they shouldn't be in the `BootstrapOptions` type?

Reply:
Options in BootstrapOptions can be required or optionals.    I don't think BootstrapOptions implies all of the fields are optionals.   We could rename it to BootstrapParameters ....

Reply:
Renaming it sounds good, certainly when I conceived of the idea, I intended for the main driver function to accept required parameters as args, and optional things as an options type argument.

We've shifted away from this, but options still remains, which feels odd.

Reply:
Renamed.